### PR TITLE
feat(gnu): add version string comparison algorithm

### PR DESCRIPTION
**Pull request by MeteorsLiu**
Implement GNU-style version comparison (strverscmp) for sorting version strings naturally. The algorithm:
- Compares numeric segments by value, not lexicographically
- Handles leading zeros correctly
- Supports tilde (~) prefix sorting (sorts before empty)
- Maintains proper ordering of letters vs special characters

Includes comprehensive test suite covering edge cases like multi-part versions, alphanumeric strings, and real-world examples.

### CODE REVIEW

File: pkgs/gnu/sort_test.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
+package gnu
+
+import (
+	"testing"
+)
+
+func TestCompare(t *testing.T) {
```
Comment (luoliwoshang) [inline pkgs/gnu/sort_test.go]: maybe we can add this https://github.com/coreutils/gnulib/blob/master/lib/filevercmp.c test case?

Comment (MeteorsLiu) [reply pkgs/gnu/sort_test.go]: It seems that isn’t necessary. We only need to compare the version string. Since filevercmp.c is meant for comparing filenames that contain version information, we don’t need that here.

Comment (luoliwoshang) [reply pkgs/gnu/sort_test.go]: ok,i understand
